pkg/controller/servicemesh/webhooks/mutation/controlplane_test.go

```diff
@@ -26,8 +27,29 @@ func TestDeletedControlPlaneIsAlwaysAllowed(t *testing.T) {
 	assert.DeepEquals(response, acceptWithNoMutation, "Expected mutator to accept deleted ServiceMeshControlPlane", t)
 }
 
-func TestControlPlaneOutsideWatchedNamespaceIsAlwaysAllowed(t *testing.T) {
-	controlPlane := newControlPlane("my-smcp", "not-watched")
+func TestDeletedV2ControlPlaneIsAlwaysAllowed(t *testing.T) {
+	controlPlane := newControlPlaneV2("my-smcp", "istio-system")
+	controlPlane.Spec.Version = ""
+	controlPlane.Spec.Template = ""
+	controlPlane.DeletionTimestamp = now()
+
+	mutator, _, _ := createControlPlaneMutatorTestFixture()
+	response := mutator.Handle(ctx, newCreateRequest(controlPlane))
+	assert.DeepEquals(response, acceptWithNoMutation, "Expected mutator to accept deleted ServiceMeshControlPlane", t)
+}
```

Comment:
Can this and `TestDeletedV1ControlPlaneIsAlwaysAllowed()` be turned into a single test with two test cases?

Reply:
done.  moved all v1/v2 specific test functions into single test function with multiple cases.